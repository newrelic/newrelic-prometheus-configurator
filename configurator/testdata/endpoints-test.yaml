kubernetes:
  jobs:
    - job_name_prefix: kubernetes-job
      target_kind:
        endpoints: true

      # Added extra label to test that we append properly already existing
      # relabel configs.
      extra_relabel_config:
      - action: drop
        regex: (.+)
        source_labels:
        - test_label

newrelic_remote_write:
  license_key: nrLicenseKey
