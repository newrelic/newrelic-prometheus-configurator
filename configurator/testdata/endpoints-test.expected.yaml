scrape_configs:
  - job_name: kubernetes-job-endpoints
    honor_labels: true
    kubernetes_sd_configs:
      - role: endpoints
    relabel_configs:
      - source_labels: [ __meta_kubernetes_service_annotation_prometheus_io_scrape ]
        action: keep
        regex: "true"
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape_slow]
        action: drop
        regex: "true"
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scheme]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [ __address__, __meta_kubernetes_service_annotation_prometheus_io_port ]
        action: replace
        regex: ([^:]+)(?::\d+)?;(\d+)
        target_label: __address__
        replacement: $1:$2

      - source_labels: [__scheme__,__address__,__metrics_path__]
        action: replace
        regex: (.+);(.+);(.+)
        replacement: $1://$2$3
        target_label: scrapedTargetURL
      - source_labels: [__meta_kubernetes_endpoints_name]
        action: replace
        target_label: targetName
      - source_labels: [job]
        action: replace
        target_label: scrappedTargetKind
      - action: labeldrop
        regex: job
      - action: labelmap
        regex: __meta_kubernetes_pod_label_(.+)
        replacement: label_$1
      - source_labels: [__meta_kubernetes_namespace]
        action: replace
        target_label: namespaceName
      - source_labels: [__meta_kubernetes_service_name]
        action: replace
        target_label: serviceName
      - source_labels: [__meta_kubernetes_pod_phase]
        action: drop
        regex: Pending|Succeeded|Failed|Completed
        # Extra relabel config
      - source_labels: [ test_label ]
        action: drop
        regex: (.+)

remote_write:
  - url: https://metric-api.newrelic.com/prometheus/v1/write
    authorization:
      credentials: nrLicenseKey
